### PR TITLE
refactor(core): clear `@ts-ignore` from workflow and core packages

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

**Review by netroy — DISMISSED** [review]
This is great. Thanks 👍🏽

**Comment by netroy** [comment]
Should we merge this after the fix release is out later today?
I don't see a breaking change here, so I'm okay either way. It's your call to merge it now, or on Monday.

**Comment by ivov** [comment]
I'll wait for Monday.

**Review by netroy — DISMISSED** [review]

File: packages/core/src/NodeExecuteFunctions.ts
Comment (netroy) [inline packages/core/src/NodeExecuteFunctions.ts:285]: We should change this to either
`Object.assign(requestObject.body, requestObject.form)` or `requestObject.body = Object.assign({}, requestObject.body, requestObject.form)`

we should also perhaps (eventually) create a linting rule for `Object.assign` should be used.

File: packages/core/src/NodeExecuteFunctions.ts
Comment (netroy) [inline packages/core/src/NodeExecuteFunctions.ts:1183]: I'd rather change this to `// @ts-expect-error @TECH_DEBT: get rid of the request library` 😅

**Review by netroy — DISMISSED** [review]
besides the `Object.assign` stuff, lgtm

File: packages/core/src/NodeExecuteFunctions.ts
Comment (ivov) [reply packages/core/src/NodeExecuteFunctions.ts:285]: This pattern occurs very frequently elsewhere. Also okay if we fix this in full but separately through the lint rule to be written?

Comment (netroy) [reply packages/core/src/NodeExecuteFunctions.ts:285]: okay

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.201.0`

**Pull request merged**
